src/Symfony/Component/Security/Core/Authentication/Provider/DaoAuthenticationProvider.php

```diff
@@ -61,7 +61,7 @@ protected function checkAuthentication(UserInterface $user, UsernamePasswordToke
                 throw new BadCredentialsException('The presented password is invalid.');
             }
 
-            if ($this->userProvider instanceof PasswordUpgraderInterface && method_exists($encoder, 'needsRehash') && $encoder->needsRehash($user->getPassword())) {
+            if ($this->userProvider instanceof PasswordUpgraderInterface && method_exists($encoder, 'needsRehash') && null !== $user->getPassword() && $encoder->needsRehash($user->getPassword())) {
```

Comment:
This change shouldn't be necessary once #34779 has been merged up if I don't miss anything.

Reply:
Looks like you're right! :)